odoo/orm/models.py

```diff
@@ -3938,6 +3941,15 @@ def write(self, vals: ValuesType) -> typing.Literal[True]:
 
         if self._check_company_auto:
             self._check_company(list(vals))
+        if (cache_name := self._clear_cache_name) and any(self._ids) and self._should_clear_cache_on_fields(vals):
+            self.env.flush_all()
+            self.env.registry.clear_cache(cache_name)
+        return True
+
+    @api.model
+    def _should_clear_cache_on_fields(self, field_names):
```

Comment:
TBD: we have many caches
`env.transaction.cache`, `env.cr.cache`, `env.registry.__caches`
maybe use `ormcache` as explicit possible?